core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ZoneService.java

```diff
@@ -74,14 +75,16 @@
     return Optional.ofNullable(zone);
   }
 
-  public Optional<Zone> upsert(Zone zone) throws ValidationException {
-    return !read(zone.getKey()).isPresent() ? create(zone) : update(zone);
+  @PostAuthorize("returnObject.isEmpty() ? true: hasPermission(returnObject, 'READ')")
+  public Optional<Zone> get(ZoneKey key) {
```

Comment:
Not sure how I feel about maintaining both `get` (authz) and `find` (no-authz) methods. If it's specifically to bypass authz then I think they should be named as such so it's clear. However, I think it's ok to only have one method (`get`) on the basis that you need to be able to READ an entity to also be able to UPDATE it.

Reply:
I will rename the methods appropriately.

If we maintain only one method (for get and read), then RangerAuthoriser will get two calls for any update or update_status (one for get and one for update). Also validators will result in further RangerAuthoriser calls.